agent/agentcontainers/dcspec/gen.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script requires quicktype to be installed.
+# While you can install it using npm, we have it in our devDependencies
+# in ${PROJECT_ROOT}/package.json.
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+if ! pnpm list | grep quicktype &>/dev/null; then
+	echo "quicktype is required to run this script!"
+	echo "Ensure that it is present in the devDependencies of ${PROJECT_ROOT}/package.json and then run pnpm install."
+	exit 1
+fi
+
+DEST_FILENAME="dcspec_gen.go"
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+DEST_PATH="${SCRIPT_DIR}/${DEST_FILENAME}"
+
+# Location of the JSON schema for the devcontainer specification.
+SCHEMA_SRC="https://raw.githubusercontent.com/devcontainers/spec/refs/heads/main/schemas/devContainer.base.schema.json"
+SCHEMA_DEST="${SCRIPT_DIR}/devContainer.base.schema.json"
+
+UPDATE_SCHEMA="${UPDATE_SCHEMA:-false}"
+if [[ "${UPDATE_SCHEMA}" = true || ! -f "${SCHEMA_DEST}" ]]; then
+	# Download the latest schema.
+	echo "Updating schema..."
+	curl --fail --silent --show-error --location --output "${SCHEMA_DEST}" "${SCHEMA_SRC}"
+else
+	echo "Using existing schema..."
+fi
+
+TMPDIR=$(mktemp -d)
+trap 'rm -rfv "$TMPDIR"' EXIT
+pnpm exec quicktype \
+	--src-lang schema \
+	--lang go \
+	--just-types-and-package \
+	--top-level "DevContainer" \
+	--out "${TMPDIR}/${DEST_FILENAME}" \
+	--package "dcspec" \
+	"${SCHEMA_DEST}"
+
+# Format the generated code.
+go run mvdan.cc/gofumpt@v0.4.0 -w -l "${TMPDIR}/${DEST_FILENAME}"
+mv -v "${TMPDIR}/${DEST_FILENAME}" "${DEST_PATH}"
+
+# Add a header so that Go recognizes this as a generated file.
+sed -i '1s/^/\/\/ Code generated by dcspec\/gen.sh. DO NOT EDIT.\n/' "${DEST_PATH}"
```

Comment:
This transformation could be done in temp still, before the move.

But unfortunately, this is not cross-platform 😔.

If you want a quick fix and not use another tool, see: https://github.com/coder/coder/commit/a1f5468db2bedcd627a44e80c31e515fc70ef3f2

Reply:
Nice callout, thanks!